chia/wallet/nft_wallet/nft_wallet.py

```diff
@@ -930,3 +934,294 @@ async def set_nft_did(self, nft_coin_info: NFTCoinInfo, did_id: bytes, fee: uint
+        # puzzle based on the starting edition number given, and the size of the
+        # chunk going into this spend bundle
+        mint_number_end = mint_number_start + chunk_size
+
+        # Empty set to load with the announcements we will assert from DID to
+        # match the announcements from the intermediate launcher puzzle
+        did_announcements: Set = set()
+        puzzle_assertions: Set = set()
+        amount = uint64(1)
+        intermediate_coin_spends = []
+        launcher_spends = []
+        launcher_ids = []
+        eve_spends = []
+        p2_inner_puzzle = await self.standard_wallet.get_new_puzzle()
+        p2_inner_ph = p2_inner_puzzle.get_tree_hash()
+
+        # Loop to create each intermediate coin, launcher, eve and (optional) transfer spends
+        for mint_number in range(mint_number_start, mint_number_end):
+            # Create  the puzzle, solution and coin spend for the intermediate launcher
+            intermediate_launcher_puz = did_wallet_puzzles.DID_NFT_LAUNCHER_MOD.curry(
+                did_wallet_puzzles.LAUNCHER_PUZZLE_HASH, mint_number, mint_total
+            )
+            intermediate_launcher_ph = intermediate_launcher_puz.get_tree_hash()
+            primaries.append(
+                AmountWithPuzzlehash(
+                    {
+                        "puzzlehash": intermediate_launcher_ph,
+                        "amount": uint64(0),
+                        "memos": [intermediate_launcher_ph],
+                    }
+                )
+            )
+            intermediate_launcher_sol = Program.to([])
+            intermediate_launcher_coin = Coin(did_coin.name(), intermediate_launcher_ph, uint64(0))
+            intermediate_launcher_coin_spend = CoinSpend(
+                intermediate_launcher_coin, intermediate_launcher_puz, intermediate_launcher_sol
+            )
+            intermediate_coin_spends.append(intermediate_launcher_coin_spend)
+
+            # create an ASSERT_COIN_ANNOUNCEMENT for the DID spend. The
+            # intermediate launcher coin issues a CREATE_COIN_ANNOUNCEMENT of
+            # the mint_number and mint_total for the launcher coin it creates
+            intermediate_announcement_message = std_hash(int_to_bytes(mint_number) + int_to_bytes(mint_total))
+            did_announcements.add(std_hash(intermediate_launcher_coin.name() + intermediate_announcement_message))
+
+            # Create the launcher coin, and add its id to a list to be asserted in the DID spend
+            launcher_coin = Coin(intermediate_launcher_coin.name(), did_wallet_puzzles.LAUNCHER_PUZZLE_HASH, amount)
+            launcher_ids.append(launcher_coin.name())
+
+            # Grab the metadata from metadata_list. The index for metadata_list
+            # needs to be offset by mint_number_start, and since
+            # mint_number starts at 1 not 0, we also subtract 1.
+            metadata = metadata_list[mint_number - mint_number_start - 1]
```

Comment:
I think this calculation is not correct.
Yes, `mint_number_start` does start at 1, but so does `mint_number`.
So for the first element (`mint_number = 1`, `mint_number_start = 1`), `mint_number - mint_number_start - 1` results in `-1`, not `0`.